packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -118,28 +142,33 @@ class _Compiler {
         while (compilationQueue.isNotEmpty) {
           final _CompilationRequest request = compilationQueue.first;
           printTrace('Compiling ${request.path}');
+          compiler ??= createCompiler();
+          suppressOutput = false;
           final String outputPath = await compiler.recompile(request.path,
             <String>[request.path],
             outputPath: outputDill.path,
           );
-          // Copy output dill next to the source file.
-          final File kernelReadyToRun = await fs.file(outputPath).copy(
-              request.path + '.dill');
-          compiler.accept();
-          compiler.reset();
-          request.result.complete(kernelReadyToRun.path);
+
+          // Check if the compiler produced the output. It could have crashed
+          // in which case outputPath would be null. In this case pass
```

Comment:
null is produced when compiler was not able to recover from compilation errors. If compiler crashed we wouldn't receive output at all as frontend server would not write <result> tag into stdout, I don't think.

Reply:
Updated the comment.